catalyst_voices/packages/internal/catalyst_voices_blocs/lib/src/category/category_detail_cubit.dart

```diff
@@ -1,57 +1,54 @@
-import 'dart:math';
-
 import 'package:catalyst_voices_blocs/catalyst_voices_blocs.dart';
 import 'package:catalyst_voices_models/catalyst_voices_models.dart';
 import 'package:catalyst_voices_services/catalyst_voices_services.dart';
 import 'package:catalyst_voices_view_models/catalyst_voices_view_models.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
 class CategoryDetailCubit extends Cubit<CategoryDetailState> {
-  // ignore: unused_field
   final CampaignService _campaignService;
   CategoryDetailCubit(
     this._campaignService,
   ) : super(const CategoryDetailState(isLoading: true));
 
-  Future<void> getCategoryDetail(String categoryId) async {
-    emit(state.copyWith(isLoading: true));
-
-    final isSuccess = await Future.delayed(
-      const Duration(seconds: 1),
-      () => Random().nextBool(),
-    );
-
-    // TODO(LynxLynxx): get data from campaignService
-
-    emit(
-      state.copyWith(
-        category: CampaignCategoryViewModel.dummy(id: categoryId),
-        isLoading: false,
-        error: isSuccess
-            ? const Optional.empty()
-            : const Optional.of(LocalizedUnknownException()),
-      ),
-    );
-  }
-
   Future<void> getCategories() async {
     if (!state.isLoading) {
       emit(state.copyWith(isLoading: true));
     }
 
-    // TODO(LynxLynxx): get data from campaignService
+    final categories = await _campaignService.getCampaignCategories();
 
-    final categories = List.generate(
-      6,
-      (index) => CampaignCategoryViewModel.dummy(id: index.toString()),
-    );
+    final categoriesModels =
+        categories.map(DetailedCampaignCategoryViewModel.fromModel).toList();
 
     emit(
       state.copyWith(
         isLoading: false,
-        categories: categories,
+        categories: categoriesModels,
         error: const Optional.empty(),
       ),
     );
   }
+
+  void getCategoryDetail(String categoryId) {
+    emit(state.copyWith(isLoading: true));
+
+    try {
+      final category = _campaignService.getCategory(categoryId);
+      emit(
+        state.copyWith(
+          isLoading: false,
+          category: DetailedCampaignCategoryViewModel.fromModel(category),
+          error: const Optional.empty(),
+        ),
+      );
+    } on Exception catch (_) {
+      emit(
```

Comment:
You can just omit `catch` if you don't intend to consume the exception, like that:
```suggestion
    } on Exception {
      emit(
```